cas-server-core/src/main/java/org/jasig/cas/CentralAuthenticationServiceImpl.java

```diff
@@ -256,8 +257,8 @@ public ServiceTicket grantServiceTicket(
             }
             ticketGrantingTicket.getSupplementalAuthentications().add(currentAuthentication);
         }
-        
-        if (!registeredService.isSsoEnabled() && currentAuthentication == null) {
+
+        if (currentAuthentication == null && !registeredService.getAuthorizationStrategy().isServiceAuthorizedForSso(service)) {

```

Comment:
Maybe I'm reading this out of order, but shouldn't the "registered service" already have been loaded for the "Service" so to check the service against the registered service seems weird to me (but I may retract this statement as I continue to read :-))